### PR TITLE
ER-1515 Rename ProgrammeLevel to ApprenticeshipLevel +semver: minor

### PR DESCRIPTION
**Pull request by mrpmorris**

### CODE REVIEW

File: src/Shared/Recruit.Vacancies.Client/Domain/Entities/ApprenticeshipLevelHelper.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+            if (Enum.IsDefined(typeof(ApprenticeshipLevel), value))
+            {
+                result = (ApprenticeshipLevel)value;
+                return true;
+            }
+            result = ApprenticeshipLevel.Unknown;
+            return false;
+        }
+
+        //TODO: PeteM - RemapFromEducationLevel
```
Comment (shomavg) [inline src/Shared/Recruit.Vacancies.Client/Domain/Entities/ApprenticeshipLevelHelper.cs]: This TODO still valid?

File: src/Employer/Employer.Web/ViewModels/Part1/Training/ConfirmTrainingViewModel.cs [hunk 2]
```diff
@@ -6,7 +6,7 @@ namespace Esfa.Recruit.Employer.Web.ViewModels.Part1.Training
     public class ConfirmTrainingViewModel
     {
         public string TrainingTitle { get; set; }
-        public ProgrammeLevel Level { get; set; }
+        public ApprenticeshipLevel Level { get; set; }
```
Comment (sachpatel) [inline src/Employer/Employer.Web/ViewModels/Part1/Training/ConfirmTrainingViewModel.cs]: Can I suggest that the property be called `ApprenticeshipLevel`

File: src/Shared/Recruit.Vacancies.Client/Domain/Entities/ApprenticeshipLevelHelper.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
+﻿using System;
+
+namespace Esfa.Recruit.Vacancies.Client.Domain.Entities
+{
+    public static class ApprenticeshipLevelHelper
+    {
+        //TODO: PeteM - TryRemapFromEducationLevel
```
Comment (sachpatel) [inline src/Shared/Recruit.Vacancies.Client/Domain/Entities/ApprenticeshipLevelHelper.cs]: Remove please

File: src/Shared/Recruit.Vacancies.Client/Domain/Entities/IApprenticeshipProgramme.cs [hunk 4]
```diff
@@ -9,7 +9,7 @@ public interface IApprenticeshipProgramme
         string Title { get; }
         DateTime? EffectiveFrom { get; }
         DateTime? EffectiveTo { get; }
-        ProgrammeLevel Level { get; }
+        ApprenticeshipLevel Level { get; }
```
Comment (sachpatel) [inline src/Shared/Recruit.Vacancies.Client/Domain/Entities/IApprenticeshipProgramme.cs]: If property name can be changed as well please to `ApprenticeshipLevel`

File: src/Shared/Recruit.Vacancies.Client/Infrastructure/Services/VacancySummariesProvider/VacancySummaryAggQueryResponseDto.cs [hunk 5]
```diff
@@ -33,7 +33,7 @@ internal class VacancySummaryDetails
         public DurationUnit DurationUnit { get; set; }
         public string TrainingTitle { get; set; }
         public TrainingType TrainingType { get; set; }
-        public ProgrammeLevel TrainingLevel { get; set; }
+        public ApprenticeshipLevel TrainingLevel { get; set; }
```
Comment (sachpatel) [inline src/Shared/Recruit.Vacancies.Client/Infrastructure/Services/VacancySummariesProvider/VacancySummaryAggQueryResponseDto.cs]: This one is fine 👍

File: src/Shared/UnitTests/Vacancies.Client/Domain/Entities/ApprenticeshipLevelHelperTests.cs [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+            {
+                ApprenticeshipLevelHelper.RemapFromInt(intValueToConvert);
+            });
+        }
+
+        [Fact]
+        public void TryRemapFromInt_ShouldSucceedAndReturnHigher_WhenConvertingFromFoundation5()
+        {
+            bool success = ApprenticeshipLevelHelper.TryRemapFromInt(5, out ApprenticeshipLevel result);
+            success.Should().Be(true);
```
Comment (sachpatel) [inline src/Shared/UnitTests/Vacancies.Client/Domain/Entities/ApprenticeshipLevelHelperTests.cs]: Thanks for using Fluent, just a tip `.Be(true)` can be replaced with `BeTrue()`

File: src/Employer/Employer.Web/ViewModels/VacancyPreview/VacancyPreviewViewModel.cs [hunk 7]
```diff
@@ -114,7 +114,7 @@ private bool IsEditRequired(string fieldIdentifier)
                 }.Count(s => s == VacancyPreviewSectionState.Incomplete || s == VacancyPreviewSectionState.InvalidIncomplete);
 
         public string IncompleteRequiredSectionText => "section".ToQuantity(IncompleteRequiredSectionCount, ShowQuantityAs.None);
-        public ProgrammeLevel Level { get; set; }
+        public ApprenticeshipLevel Level { get; set; }
```
Comment (shomavg) [inline src/Employer/Employer.Web/ViewModels/VacancyPreview/VacancyPreviewViewModel.cs]: Can we keep the name consistent to ApprenticeshipLevel?

File: src/Provider/Provider.Web/ViewModels/VacancyPreview/VacancyPreviewViewModel.cs [hunk 8]
```diff
@@ -114,7 +114,7 @@ public class VacancyPreviewViewModel : DisplayVacancyViewModel
         public bool VacancyDescriptionRequiresEdit => IsEditRequired(FieldIdentifiers.VacancyDescription);
         public bool WageRequiresEdit => IsEditRequired(FieldIdentifiers.Wage);
         public bool WorkingWeekRequiresEdit => IsEditRequired(FieldIdentifiers.WorkingWeek);
-        public ProgrammeLevel Level { get; set; }
+        public ApprenticeshipLevel Level { get; set; }
```
Comment (shomavg) [inline src/Provider/Provider.Web/ViewModels/VacancyPreview/VacancyPreviewViewModel.cs]: Change name to ApprenticeshipLevel.

File: src/QA/QA.Web/ViewModels/ReviewViewModel.cs [hunk 9]
```diff
@@ -121,6 +121,6 @@ private string GetFieldIdentifierCssClass(string fieldIdentifer)
         {
             return FieldIdentifiers.Single(f => f.FieldIdentifier == fieldIdentifer).FieldValueHasChanged ? CssFieldChanged : null;
         }
-        public ProgrammeLevel Level { get; set; }
+        public ApprenticeshipLevel Level { get; set; }
```
Comment (shomavg) [inline src/QA/QA.Web/ViewModels/ReviewViewModel.cs]: Change name to ApprenticeshipLevel

File: src/Shared/Recruit.Shared.Web/ViewModels/VacancySummaryViewModel.cs [hunk 10]
```diff
@@ -16,7 +16,7 @@ public class VacancySummaryViewModel
         public string ProgrammeId { get; set; }
         public string TrainingTitle { get; set; }
         public TrainingType TrainingType { get; set; }
-        public ProgrammeLevel TrainingLevel { get; set; }
+        public ApprenticeshipLevel TrainingLevel { get; set; }
```
Comment (shomavg) [inline src/Shared/Recruit.Shared.Web/ViewModels/VacancySummaryViewModel.cs]: Change name to ApprenticeshipLevel

File: src/Shared/Recruit.Vacancies.Client/Infrastructure/QueryStore/Projections/VacancySummary/VacancySummary.cs [hunk 11]
```diff
@@ -25,7 +25,7 @@ public class VacancySummary
         public DateTime? StartDate { get; set; }
         public string TrainingTitle { get; set; }
         public TrainingType TrainingType { get; set; }
-        public ProgrammeLevel TrainingLevel { get; set; }
+        public ApprenticeshipLevel TrainingLevel { get; set; }
```
Comment (shomavg) [inline src/Shared/Recruit.Vacancies.Client/Infrastructure/QueryStore/Projections/VacancySummary/VacancySummary.cs]: Change name to ApprenticeshipLevel

File: src/Shared/Recruit.Vacancies.Client/Infrastructure/Services/VacancySummariesProvider/VacancySummaryAggQueryResponseDto.cs (hunk 5 above)
Comment (shomavg) [reply src/Shared/Recruit.Vacancies.Client/Infrastructure/Services/VacancySummariesProvider/VacancySummaryAggQueryResponseDto.cs]: Change name to ApprenticeshipLevel

**Pull request merged**